api4/channel.go

```diff
@@ -1202,6 +1202,11 @@ func removeChannelMember(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if channel.GroupConstrained != nil && *channel.GroupConstrained {
```

Comment:
Wouldn't this block the possibility of a member leaving him/herself?

Reply:
I didn't think about that! Putting this back as WIP. I'll re-request a review when it's ready.